feat: added functionality in 'add members in group' like 'add collaborators in project'

vedant-jain03 opened the pull request:
Fixes #2562 

#### Describe the changes you have made in this PR -
- added CDN link of bootstrap.tagsinput
- added the functionality similar to 'add collaborator'.

### Screenshots of the changes (If any) -
- before
![bandicam 2021-10-20 12-16-14-999](https://user-images.githubusercontent.com/76901313/138139476-f2129f9f-37d2-4bbc-80f6-b7313709cd81.jpg)

- after
![Screenshot (165)](https://user-images.githubusercontent.com/76901313/138107052-04934eef-313c-47e1-839f-0d6e65a6086d.png)

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

vedant-jain03 commented:
@pavanjoshi914 @tachyons CI test cases passed. Kindly Review

tachyons commented on app/assets/stylesheets/application.scss:
```diff
@@ -13,7 +14,7 @@
 @import 'about';
 @import 'contribute';
-@import 'teacher';
+@import 'teacher';  
```
remove extra space here

vedant-jain03 replied:
@tachyons Done sir!

pavanjoshi914 commented:
Please make code climate happy!

vedant-jain03 commented on app/javascript/controllers/addMember_controller.js:
```diff
@@ -0,0 +1,57 @@
+            $('.bootstrap-tagsinput input').attr('maxlength', '30');
+            $('.bootstrap-tagsinput input').attr('id', 'group_email_input');
+            $('.add-members-button').attr('disabled', true);
+            $('.bootstrap-tagsinput input').on('keyup', () => {
```
This block of code is for 'keyup' eventlistener.

vedant-jain03 commented on app/javascript/controllers/addMember_controller.js:
```diff
@@ -0,0 +1,57 @@
+                    $('.add-members-button').attr('disabled', true);
+                }
+            });
+            $('.bootstrap-tagsinput').on('click', () => {
```
This block of code is for 'click' eventlistener.

vedant-jain03 commented:
@pavanjoshi914 Code Climate almost got happy.
@tachyons

pavanjoshi914 commented:
Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code

pavanjoshi914 commented:
```package.json``` is already pushed by you for dependencies. no need of committing ```yarn.lock```

vedant-jain03 commented:
> Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code

Actually Both functions are independent of each other. One is for 'click' event and second one is for 'keyup' event but the conditional loop is same for both of them.

vedant-jain03 commented:
> `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`

okhay I have removed package.json changes.

pavanjoshi914 commented:
> > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> 
> okhay I have removed package.json changes.

You have to remove yarn.lock. package.json is required

vedant-jain03 commented:
> > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > 
> > 
> > okhay I have removed package.json changes.
> 
> You have to remove yarn.lock that is what I said. package.json is required

Actually I have install both dependencies using yarn only that is why I have removed package.json.

pavanjoshi914 commented:
> > > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > > 
> > > 
> > > okhay I have removed package.json changes.
> > 
> > 
> > You have to remove yarn.lock that is what I said. package.json is required
> 
> Actually I think may be yarn.lock file is important as I have setup stimulus using webpacker and webpacker has installed yarn files that is important for the process. For tagsinput I have used yarn.

dependencies are preserved in ```package.json``` lock files are autogenerated

vedant-jain03 commented:
> > > > > `package.json` is already pushed by you for dependencies. no need of committing `yarn.lock`
> > > > 
> > > > 
> > > > okhay I have removed package.json changes.
> > > 
> > > 
> > > You have to remove yarn.lock that is what I said. package.json is required
> > 
> > 
> > Actually I think may be yarn.lock file is important as I have setup stimulus using webpacker and webpacker has installed yarn files that is important for the process. For tagsinput I have used yarn.
> 
> dependencies are preserved in `package.json` lock files are autogenerated

I have removed `yarn.lock` file.

pavanjoshi914 commented:
> > Is it possible to use jquery multiple class selectors in this case? to avoid duplication of code
> 
> Actually Both functions are independent of each other. One is for 'click' event and second one is for 'keyup' event but the conditional loop is same for both of them.

Yes, we can trigger multiple on the same function, can you try this 

https://stackoverflow.com/questions/2534089/jquery-multiple-events-to-trigger-the-same-function

tachyons commented:
> dependencies are preserved in package.json lock files are autogenerated

True, but not all packages follow proper versioning, So it is important to commit yarn.lock when pushing code

tachyons commented on app/javascript/controllers/addMember_controller.js:
```diff
@@ -0,0 +1,55 @@
+import 'bootstrap/dist/js/bootstrap';
+
+export default class extends Controller {
+    connect() {
```
Can you consider refactoring using  data-action https://stimulus.hotwired.dev/handbook/building-something-real

vedant-jain03 replied:
Sure Sir.

pavanjoshi914 commented:
> > dependencies are preserved in package.json lock files are autogenerated
> 
> True, but not all packages follow proper versioning, So it is important to commit yarn.lock when pushing code

@vedant-jain03  can you commit ```yarn.lock``` again

vedant-jain03 commented:
@tachyons @pavanjoshi914 Kindly Review
1. added yarn.lock
2. Refactor the code using data-action (click) for addMember button in show.html.erb .

pavanjoshi914 commented:
> @tachyons @pavanjoshi914 Kindly Review
> 
> 1. added yarn.lock
> 2. Refactor the code using data-action (click) for addMember button in show.html.erb .

Great work @vedant-jain03 !!  There are lot of commits out there can you make commit history a little clean with proper commit message. 

I suggest following commitlint convention as a good practice

vedant-jain03 commented:
> > @tachyons @pavanjoshi914 Kindly Review
> > 
> > 1. added yarn.lock
> > 2. Refactor the code using data-action (click) for addMember button in show.html.erb .
> 
> Great work @vedant-jain03 !! There are lot of commits out there can you make commit history a little clean with proper commit message.
> 
> I suggest following commitlint convention as a good practice

Thank you @pavanjoshi914 for the recommendation, I have edited my last commit message, actually stimulus was new for me and I have faced rspec testcases for the first time that is why it took me many commits. But learned a lot from every commit. 
I will try to have clean commit history from next time.

nitin10s reviewed (approved):
@tachyons
